### PR TITLE
Refactor page structure

### PR DESCRIPTION
**Pull request by Steffengreiner**
**What was changed**

1.) Refactor of the abstract main component, it now serves as a starting point of adding a new page without content and support components but enforces the resolution of a beforeenterevent to ensure the user has access to the page. 
2.) Components within support and content components were moved into their relevant main components (Biggest change in samplemain component)
3.) PageArea CSS was completely overhauled to remove the card design for each pagearea
3.) Rest of CSS was adapted so it looks the same as before with a lot of cleanup
4.) Ensure that the components are usable on smaller screen sizes

### CODE REVIEW

File: project-management/src/main/java/life/qbic/projectmanagement/domain/model/sample/Sample.java [hunk 1]
```diff
@@ -108,10 +108,17 @@ public SampleCode sampleCode() {
     return this.sampleCode;
   }
 
-  public SampleOrigin sampleOrigin() { return this.sampleOrigin; }
```
Comment (wow-such-code) [inline project-management/src/main/java/life/qbic/projectmanagement/domain/model/sample/Sample.java:111]: please revert the code styling changes in this class

File: project-management/src/main/java/life/qbic/projectmanagement/domain/service/SampleDomainService.java [hunk 2]
```diff
@@ -67,7 +67,7 @@ public void updateSamples(Project project,
                     sampleUpdateRequest -> sampleUpdateRequest.sampleId().equals(sample.sampleId()))
                 .findFirst().orElseThrow();
             sample.setLabel(sampleInfo.sampleInformation().sampleLabel());
-            sample.setOrganismId(sampleInfo.sampleInformation().organismId());
```
Comment (wow-such-code) [inline project-management/src/main/java/life/qbic/projectmanagement/domain/service/SampleDomainService.java:70]: please revert the code styling changes in this class

File: user-interface/src/main/java/life/qbic/datamanager/views/projects/project/samples/BatchDetailsComponent.java [hunk 3]
```diff
@@ -119,6 +121,7 @@ private void createBatchGrid() {
         .setHeader("Action");
     batchGrid.addThemeVariants(GridVariant.LUMO_COMPACT);
     batchGrid.addClassName("batch-grid");
+    batchGrid.setAllRowsVisible(true);
```
Comment (wow-such-code) [inline user-interface/src/main/java/life/qbic/datamanager/views/projects/project/samples/BatchDetailsComponent.java:124]: why is this necessary?

File: user-interface/src/main/java/life/qbic/datamanager/views/projects/project/samples/registration/batch/SampleBatchInformationSpreadsheet.java [hunk 4]
```diff
@@ -212,7 +212,11 @@ public String getSampleLabel() {
     public void setSampleLabel(String sampleLabel) {
       this.sampleLabel = sampleLabel;
     }
-    public String getOrganismId() { return organismId; }
```
Comment (wow-such-code) [inline user-interface/src/main/java/life/qbic/datamanager/views/projects/project/samples/registration/batch/SampleBatchInformationSpreadsheet.java:215]: please revert the code styling changes in this class

Comment (wow-such-code) [reply user-interface/src/main/java/life/qbic/datamanager/views/projects/project/samples/registration/batch/SampleBatchInformationSpreadsheet.java:215]: ...for clarity

File: user-interface/src/main/java/life/qbic/datamanager/views/projects/project/samples/BatchDetailsComponent.java (hunk 3 above)
Comment (Steffengreiner) [reply user-interface/src/main/java/life/qbic/datamanager/views/projects/project/samples/BatchDetailsComponent.java:124]: This enables the grid to be dynamic in size dependent on the number of batches. 
This is intended to decrease the space the batch component will take on the sample page since we can expect a less number of batches for an experiment normally than is the default height of 400px
https://vaadin.com/docs/latest/components/grid#dynamic-height

File: user-interface/src/main/java/life/qbic/datamanager/views/projects/project/samples/registration/batch/SampleBatchInformationSpreadsheet.java (hunk 4 above)
Comment (Steffengreiner) [reply user-interface/src/main/java/life/qbic/datamanager/views/projects/project/samples/registration/batch/SampleBatchInformationSpreadsheet.java:215]: I reformated the file according to google styling and removed the custom linebreaks

File: project-management/src/main/java/life/qbic/projectmanagement/domain/model/sample/Sample.java (hunk 1 above)
Comment (KochTobi) [reply project-management/src/main/java/life/qbic/projectmanagement/domain/model/sample/Sample.java:111]: no the change is correct according to our style guide :) 
```
void something() {
}
``` 
would be valid.

**Review by wow-such-code — APPROVED** [review]
